xarray/backends/zarr.py

```diff
@@ -615,9 +632,27 @@ def open_store_variable(self, name, zarr_array=None):
         encoding = {
             "chunks": zarr_array.chunks,
             "preferred_chunks": dict(zip(dimensions, zarr_array.chunks, strict=True)),
-            "compressor": zarr_array.compressor,
-            "filters": zarr_array.filters,
         }
+
+        if _zarr_v3() and zarr_array.metadata.zarr_format == 3:
+            encoding["codec_pipeline"] = [
+                x.to_dict() for x in zarr_array.metadata.codecs
```

Comment:
Maybe this instead?

```suggestion
                zarr_array.metadata.to_dict()["codecs"]
```

A bit wasteful since everything has to be serialized, but presumably zarr knows better how to serialize the codec pipeline than we do here?